File: abc/abc178/b.cpp
// https://atcoder.jp/contests/abc178/submissions/27577004

#include <bits/stdc++.h> 
using namespace std; 
int main() {
  long long a, b, c, d;
  cin >> a >> b >> c >> d;
  cout << max({a*c, a*d, b*c, b*d});
}

File: abc/abc178/b.py
# https://atcoder.jp/contests/abc178/submissions/38147065

a, b, c, d = map(int, input().split())
x = [a, b]
y = [c, d]

ans = -1e19
for i in x:
  for j in y:
    ans = max(ans, i * j)

print(ans)
